### PR TITLE
Latest master -> Dev branch merge, with the new gamepad work

### PR DESCRIPTION
**Pull request by keveleigh**

### CODE REVIEW

File: ProjectSettings/InputManager.asset [hunk 1]
```diff
@@ -294,7 +294,7 @@ InputManager:
     axis: 0
     joyNum: 0
   - serializedVersion: 3
-    m_Name: ControllerLeftStickX
```
Comment (StephenHodgson) [inline ProjectSettings/InputManager.asset]: @keveleigh did you get to test this with the motion controllers?

I still plan on revisiting this to make sure we're clear of any conflicts.

File: Packages/manifest.json [hunk 2]
```diff
@@ -0,0 +1,4 @@
+{
+	"dependencies": {
```
Comment (StephenHodgson) [inline Packages/manifest.json]: Should we ignore or include this in our project files?

Comment (keveleigh) [reply Packages/manifest.json]: Ahh, this slipped past. I'm not sure...it seems like recent Unity versions are generating this, but I don't think we'd need it unless we use it. I can add it to the .gitignore.

**Comment by keveleigh** [comment]
@StephenHodgson I'll double check.

**Comment by DDReaper** [comment]
Somethings not right, it's reverted the project version to 2017.2.0b11
SO it's missing/overwritten the commit for updating the project in #1015

**Comment by keveleigh** [comment]
Thanks for catching!

File: Packages/manifest.json (hunk 2 above)
Comment (StephenHodgson) [reply Packages/manifest.json]: Yeah, I'm unsure exactly what these dependencies are or what they're used for.

**Comment by StephenHodgson** [comment]
Besides the comments above LGTM

**Comment by StephenHodgson** [comment]
Didn't mean to request changes, just comment.

**Comment by keveleigh** [comment]
@StephenHodgson It was moved out of Examples in this branch by @darax to put it side-by-side with the original Sharing folder, since it's a feature area on par with that one.

**Comment by StephenHodgson** [comment]
Then why does it have the example assets in the core folder?

**Comment by StephenHodgson** [comment]
I'm for adding it in as a core feature, but it seems more like an example than just a base asset.

**Comment by StephenHodgson** [comment]
I think we should match up the folder structure as closely as possible for now, so we can do a clean comparison and add it back in later.

**Comment by StephenHodgson** [comment]
Both Xbox and Motion controllers worked for me in the Motion Controller test scene.  Great work!

**Comment by jessemcculloch** [comment]
Looks like a bunch of the input manager tests are failing.  Maybe need to update the tests?

image

**Comment by jessemcculloch** [comment]
It looks like the FocusManager singleton instance is returning null for all of them

**Comment by StephenHodgson** [comment]
@jessemcculloch, I think those tests were designed for the older input system.  It'll need to be updated.

**Comment by StephenHodgson** [comment]
Let's worry about that in another PR

**Pull request merged**